### DISCUSSION TITLE
Installing/running concourse on a remote server

### DISCUSSION THREAD

**Discussion by folibis** [concourse/concourse#6958] [General]
I want to replace drone.io that runs on out remote server with concourse. So I follow the official tutorial: https://concourse-ci.org/quick-start.html to try the application.

The only lines I've replaced in docker-compose.yml are:

```
ports: ["8081:8081"]
CONCOURSE_EXTERNAL_URL: http://myserver.com:8081
```
since 8080 is required be drone.io instance. So I run that by  ```docker-compose up``` and I see lots of logs. But I cannot connect to the web application. The tutorial proposes to connect to http://localhost:8080 but since I've changed it I connect to http://myserver.com:8081. Nothing happens. The server is available, the firewall is off but I still can't connect. 
The netstat shows the listen connection:

```
tcp6       0      0 :::8081                 :::*                    LISTEN
```
For some reason it runs as ipv6, I guess I need ipv4 but I don't know how to configure that.

Then I've tried to run the application following this tutorial: https://concourse-ci.org/install.html

I've set up the postgresql, ssh keys, environment variables etc. but after running the application:
```
/usr/local/bin/concourse web
```
I get error: 

> error: the required flag `--tsa-host-key' was not specified

So how can I run the application?

**Comment by patmaddox** [comment]
I don't know about the `--tsa-host-key` flag.

It may be that Concourse is still running on 8080, and so you need to map 8081 from the host to 8080 of the container? Which I believe would be `ports: ["8081:8080"]`.

**Comment by folibis** [comment]
As I said I have another application (drone.io) running on 8080 so I've configured (I guess I did) Concourse to be run on 8081. Actually I didn't find any doc that describe what and how can I remap the ports. Btw if I run a standalone application I guess it doesn't run inside a container.

**Comment by folibis** [comment]
Ok, solved by setting the flags explicitly:
```
/usr/local/bin/concourse web --tsa-host-key=/etc/concourse/tsa_host_key --add-local-user=admin:admin --main-team-local-user=admin --postgres-user=concourse --postgres-password=*** --bind-port=8081 --external-url=http://myserver.com:8081 
```
